docs/recipes/immutability-helpers.md

```diff
@@ -0,0 +1,190 @@
+# Immutability Helpers
+
+Redux is a tiny pattern that represents states as immutable objects. Redux was originally designed for React. Most Redux concepts, such as pure functions, are centered around the React ecosystem. Nowadays Redux is not directly related to React.
+
+The cornerstone of Redux is immutability. Immutabilty is an amazing pattern to minimise unpredictable behaviour in our code. We're not going to cover functional programming in this article. However we're going to look at very useful packages that are called "immutability helpers". Most developers have to deal with, so called, "deep objects" and most important follow the immutability concept, when it comes to changing the value of some deeply nested property.
+
```

Comment:
We should add our own state operators here at the top of this list with a link to the page on that.
State Operators are essentially immutibility helpers.

Also, we could add https://github.com/kolodny/immutability-helper to the list below. I have used it before and it works well. It is also quite well used for Redux.

Reply:
Good point